programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1822,12 +1823,65 @@ static struct option run_host_options[] = {
 static const char* dns_upstream = NULL;
 static bool host_only = false;
 
+#include "tlsuv/http.h"
+static struct tlsuv_url_s *proxy_url = NULL;
```

Comment:
why allocate this instead of static memory?

Reply:
when it's null it means that no proxy was specified. alternatively there could be a separate boolean, but allocating it and checking for non-null felt cleaner to me.

Reply:
you can just check proxy_url.host for non-NULL
if it wasn't specified it won't be set

also might avoid global var altogether since it is not needed after you set global connector

Reply:
the init logic has been changed to use the global connector